### PR TITLE
feat: replace pip install with uv pip install in CI workflows

### PR DESCRIPTION
**Pull request by rahulsamant37**
**Description**
This pull request updates our CI workflows to replace pip install with uv pip install for faster dependency installation. Based on testing in a fresh virtual environment, the following results were observed:

- pip install: 30 seconds
- uv pip install: 8 seconds (this includes the time to install uv itself, which takes only a few seconds)
This change leverages uv, which supports editable and local installs, improving the overall speed and efficiency of our CI processes.

Closes: #7442

**Tests**

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up your Gradio environment locally by following the [contributing guide](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests:
```
bash scripts/run_all_tests.sh
```
2. You may need to run the linters:
```
bash scripts/format_backend.sh
bash scripts/format_frontend.sh
```

### CODE REVIEW

**Comment by abidlabs** [comment]
Where is uv installed in the workflow?

File: .github/actions/install-all-deps/action.yml [hunk 1]
```diff
@@ -54,6 +54,12 @@ runs:
       run: |
         python -m pip install --upgrade virtualenv
         python -m virtualenv venv
+    - name: Install uv
+      shell: bash
+      run: |
+        . ${{ env.VENV_ACTIVATE }}
+        python -m pip install uv
```
Comment (pngwn) [inline .github/actions/install-all-deps/action.yml:61]: @abidlabs `uv` is installed here with pip. bBut we should use the standalone installer instead:

```
curl -LsSf https://astral.sh/uv/install.sh | sh
```

File: .github/actions/install-all-deps/action.yml [hunk 2]
```diff
@@ -71,14 +77,14 @@ runs:
...
+        uv python -m pip install -r test/requirements.txt
+        uv python -m pip install -r client/python/test/requirements.txt
     - name: Install Gradio and Client Libraries Locally (Linux)
       shell: bash
       run: |
         . ${{ env.VENV_ACTIVATE }}
-        python -m pip install -e client/python
-        python -m pip install -e ".[oauth]"
+        uv python -m pip install -e client/python
+        uv python -m pip install -e ".[oauth]"
```
Comment (pngwn) [inline .github/actions/install-all-deps/action.yml:87]: I think we can just change all of these to `uv pip install ...`

**Review by pngwn — COMMENTED** [review]
Few small comments. Thanks @rahulsamant37!

**Comment by rahulsamant37** [comment]
Can you tell me what is happening that is wrong now?

**Comment by abidlabs** [comment]
Will take a look soon!

**Review by abidlabs — APPROVED** [review]
lgtm, let's merge it in see if works

**Pull request merged**